crates/host/src/wasmbus/mod.rs

```diff
@@ -2328,36 +2356,68 @@ impl Host {
             .read()
             .await
             .iter()
-            .flat_map(
+            .filter_map(
                 |(
                     public_key,
                     Provider {
-                        claims, instances, ..
+                        claims,
+                        instances,
+                        image_ref,
+                        ..
                     },
                 )| {
-                    instances.keys().map(move |link_name| {
-                        let metadata = claims.metadata.as_ref();
-                        let contract_id =
-                            metadata.map(|jwt::CapabilityProvider { capid, .. }| capid.as_str());
-                        json!({
-                            "public_key": public_key,
-                            "link_name": link_name,
-                            "contract_id": contract_id.unwrap_or("n/a"),
-                        })
-                    })
+                    let jwt::CapabilityProvider {
+                        capid: contract_id,
+                        name,
+                        rev: revision,
+                        ..
+                    } = claims.metadata.as_ref()?;
```

Comment:
This should never fail, right? Was wondering if we should log if it does

Reply:
Yeah it shouldn't, I mean this would have to be a scenario where the claims had no metadata, which I think isn't possible based on how we admit claims 🤔